build_runner/lib/src/asset_graph/node.dart

```diff
@@ -74,13 +76,26 @@ class GeneratedAssetNode extends AssetNode {
 
   /// All the inputs that were read when generating this asset, or deciding not
   /// to generate it.
-  final Set<AssetId> inputs;
+  ///
+  /// This needs to be an ordered set because we compute combined input digests
+  /// using this later on.
+  final SplayTreeSet<AssetId> inputs;
+
+  /// A digest combining all digests of all previous inputs.
+  ///
+  /// This is used to determine if a node really needs to be output or not.
```

Comment:
[optional] I might rephrase this, the `Builder` still decides whether to output an asset, we decide whether to run the build.

"Used to determine whether all the inputs to a build step are identical to the previous run indicating that the previous output is still valid."